satpy/tests/test_scene.py

```diff
@@ -1796,6 +1796,35 @@ def test_resample_ancillary(self):
         new_scene = scene.resample(dst_area)
         assert new_scene['comp20'] is new_scene['comp19'].attrs['ancillary_variables'][0]
 
+    def test_resample_multi_ancillary(self):
+        """Test that multiple ancillary variables are retained after resampling.
+
+        This test corresponds to GH#2329
+        """
+        from pyresample import create_area_def
+        sc = Scene()
+        n = 5
+        ar = create_area_def("a", 4087, resolution=1000, center=(0, 0), shape=(n, n))
+        sc["test"] = xr.DataArray(
+            np.arange(n*n).reshape(n, n),
+            dims=("y", "x"),
+            attrs={
+                "area": ar,
+                "name": "test",
+                "ancillary_variables": [
+                    xr.DataArray(
+                        np.arange(n*n).reshape(n, n)*i,
+                        dims=("y", "x"),
+                        attrs={
+                            "name": f"anc{i:d}",
+                            "area": ar})
+                        for i in range(2)]})
```

Comment:
This is a bit difficult to read. I think something like this improves readability:
```
anc_vars = [xr.DataArray(np.arange(n*n).reshape(n, n)*i, dims=("y", "x"),
                         attrs={
                             "name": f"anc{i:d}",
                             "area": ar})
            for i in range(2)]

sc["test"] = xr.DataArray(np.arange(n*n).reshape(n, n), dims=("y", "x"),
                          attrs={
                              "area": ar,
                              "name": "test",
                              "ancillary_variables": anc_vars})
```

Reply:
Fixed in ad3ff7665.